openssl-dgst, openssl-genpkey: add page (#4879)
tldr-pages/tldr: pages/common/openssl-genpkey.md

# openssl genpkey

> OpenSSL command to generate asymmetric key pairs.

> > More information: <https://www.openssl.org/docs/man1.0.2/man1/genpkey.html>.

Reviewer: Can we use this link to keep the link updated?
```suggestion
> More information: <https://www.openssl.org/docs/manmaster/man1/genpkey.html>.
```

> > More information: <https://www.openssl.org/docs/manmaster/man1/genpkey.html>.

Reviewer: ```suggestion
> More information: <https://www.openssl.org/docs/manmaster/man1/openssl-genpkey.html>.
```

> - Generate a 2048bit RSA private key and save it to a file:

Reviewer: Based on `openssl-genrsa` page:
```suggestion
- Generate an RSA private key of 2048 bits, saving it to a specific file:
```

> `openssl genpkey -algorithm rsa -pkeyopt rsa_keygen_bits:2048 -out {{filename.key}}`

Reviewer: ```suggestion
`openssl genpkey -algorithm {{rsa}} -pkeyopt {{rsa_keygen_bits}}:{{2048}} -out {{filename.key}}`
```

Author: `rsa`and `rsa_keygen_bits` are not really optional here, since the command is specifically for creating an RSA key. I would just tokenize the bitlength ot `{{2048}}`

Reviewer: I'm not talking about optional arguments, I mean an option that specific argument `-algorithm` because other values are allowed.

Author: Not really. If you want to generate an RSA key, and that's what this command is about, than `rsa` is the only viable option here. It would be a completely different command otherwise.
Due to the different set of option, I consider it helpful to have separate commands for different key types here.

Reviewer: Looks the `tldr` page: the `Show the tar tldr page for Linux` example. Both, `linux` and `tar` are between `{{}}` but if you want the page for `tar` in `linux` this example is the only way to get it. but `linux` and `tar` are argument options. like `rsa` here.

Author: OK, if that's the way you want it I'll accept that.

Author: Is using one token for the option name/value pair OK?

Reviewer 2: Hrm, I see what's happened here - looks like a case of crossed wires. What about this?
```suggestion
`openssl genpkey -algorithm rsa -pkeyopt rsa_keygen_bits:{{2048}} -out {{filename.key}}`
```

@navarroaxel: Looks like this example is specifically about rsa, so other values don't make sense here because fo the extra `rsa_keygen_bits` argument.

> - Generate an elliptic curve private key using the curve *prime256v1* and save it to a file:

Reviewer: ```suggestion
- Generate an elliptic curve private key using the curve `prime256v1`, saving it to a specific file:
```

> `openssl genpkey -algorithm EC -pkeyopt ec_paramgen_curve:prime256v1 -out {{filename.key}}`

Reviewer: ```suggestion
`openssl genpkey -algorithm {{EC}} -pkeyopt {{ec_paramgen_curve}}:{{prime256v1}} -out {{filename.key}}`
```

Author: Same comment as for the RSA key generation. The only token here should be `{{prime256v1}}`

Reviewer 2: Perhaps this will read better?
```suggestion
`openssl genpkey -algorithm EC -pkeyopt ec_paramgen_curve:{{prime256v1}} -out {{filename.key}}`
```

> - Generate an ED25519 elliptic curve private key and save it to a file:

Reviewer: ```suggestion
- Generate an `ED25519` elliptic curve private key, saving it to a specific file:
```

> `openssl genpkey -algorithm ED25519 -out {{filename.key}}`

Reviewer: ```suggestion
`openssl genpkey -algorithm {{ED25519}} -out {{filename.key}}`
```

Author: Same comment as for RSA key generation. `ED25519` should not be in a token.
